### PR TITLE
UCP/DT: Fix bug in ucp_contig_dt_length usage in common places

### PR DESCRIPTION
**Pull request by shssf**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2752/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/168/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2138/ for details (Mellanox internal link).

File: src/ucp/dt/dt.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+static inline
+size_t ucp_dt_length(ucp_datatype_t datatype, size_t count,
+                     const ucp_dt_iov_t *iov, const ucp_frag_state_t *state)
+{
+    ucp_dt_generic_t *dt_gen;
+    size_t length = 0;
+
+    switch (datatype & UCP_DATATYPE_CLASS_MASK) {
+    case UCP_DATATYPE_CONTIG:
+        length = ucp_contig_dt_length(datatype, count);
```
Comment (yosefe) [inline src/ucp/dt/dt.h]: could be written simpler w/o extra variable "length" - just return the value directly from 'case' instead of doing "break"

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/180/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2765/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2151/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2152/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp pls take a look

**Comment by shamisp** [comment]
👍

**Pull request merged**